### ISSUE TITLE
YOLOE seg→det ONNX bug

### ISSUE THREAD

**Issue by jozefchutka** [ultralytics/ultralytics#22744]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi Ultralytics team,

as discussed on https://github.com/ultralytics/ultralytics/issues/22630#issuecomment-3546086901 I am creating a separate issue of the reported problem.

Using the following exported [FP16 model](https://github.com/ultralytics/ultralytics/issues/22630#issuecomment-3527924787)

```sh
!curl https://github.com/ultralytics/assets/releases/download/v8.3.0/yoloe-11s-seg.pt --output yoloe-11s-seg.pt -L
!pip install ultralytics
from ultralytics import YOLOE
from ultralytics.utils.patches import torch_load

det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")

model = YOLOE("yoloe-11s-seg-det.pt")
model.export(format="onnx", opset=17, simplify=True, dynamic=False, nms=False, device=0, half=True)
```

and running it with 

```html
<script src="https://cdn.jsdelivr.net/npm/onnxruntime-web@1.22.0/dist/ort.webgpu.min.js"></script>
<script>
(async () => {
const session = await ort.InferenceSession.create("yoloe-11s-seg-det.onnx", {executionProviders: ["webgpu"]});
const inputData = new Float16Array(1 * 3 * 640 * 640);
const inputTensor = new ort.Tensor("float16", inputData, [1, 3, 640, 640]);
const results = await session.run({images:inputTensor});
console.log(results.output0.data.slice(40000, 40010))
})()
</script>
```

This will print a slice of data into console like:

```
Float16Array(10) [NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, buffer: ArrayBuffer(20), byteLength: 20, byteOffset: 0, length: 10, Symbol(Symbol.toStringTag): 'Float16Array']
```

- tested with wasm instead of webgpu - same results.

[FP32 version](https://github.com/ultralytics/ultralytics/issues/22630#issuecomment-3542189941):

```sh
!curl https://github.com/ultralytics/assets/releases/download/v8.3.0/yoloe-11s-seg.pt --output yoloe-11s-seg.pt -L
!pip install ultralytics
from ultralytics import YOLOE
from ultralytics.utils.patches import torch_load

det = YOLOE("yoloe-11s.yaml")
state = torch_load("yoloe-11s-seg.pt")  # or use yoloe-11s-seg-pf.pt
det.load(state["model"])
det.save("yoloe-11s-seg-det.pt")

model = YOLOE("yoloe-11s-seg-det.pt")
model.export(format="onnx", opset=17, simplify=False, dynamic=False, nms=False, device=0, half=False)
``` 

does not return NaN-s but it still does not detect anything - the values are super low basically returning no detections on real image

### Environment

env: onnxruntime-web

### Minimal Reproducible Example

...provided

### Additional

Just a note that I won’t be able to assist with further debugging — I’m mainly reporting this to help surface the problem for you. Hope it’s useful.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jozefchutka, thank you for opening this issue and for your detailed report and reproducible steps 🚀  
This is an automated response to help triage your issue quickly — an Ultralytics engineer will review this thread and assist you further soon 😊

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 Bug Report related to YOLOE segmentation-to-detection conversion and ONNX export, thank you for already providing a clear workflow and environment details. If you have the possibility to further minimize it (for example, smallest script and model variant that still reproduces the NaN outputs or missing detections), that kind of [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) is extremely helpful for the engineering team.

If this evolves into a custom training or export ❓ Question (for example, about alternative export settings or configs), please share:
- dataset or input image examples (where possible)
- export commands and logs
- any runtime logs or warnings from onnxruntime-web  
and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) if training is involved.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before the team digs in, please make sure you’ve upgraded to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to confirm the behavior still occurs on the newest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If you are able to reproduce the same ONNX behavior in one of these verified environments (for example, via a Colab notebook exporting and running the same model), sharing that notebook or exact steps will make it much easier for the team to pinpoint the issue.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You will need to set the classes before exporting. Otherwise, it's not configured to detect any classes.


```python
from ultralytics import YOLOE

model = YOLOE("yoloe-11s.yaml").load("yoloe-11s-seg.pt")
model.set_classes(["person", "car"])
model.export(format="onnx", opset=17, simplify=False, dynamic=False, nms=False, device=0, half=False)
```

**Comment by jozefchutka** [comment]
I can confirm these steps worked and produced working .onnx model. Thanks for your help @Y-T-G
